src/Microsoft.OData.Core/UriParser/Resolver/ODataUriResolver.cs

```diff
@@ -99,14 +100,23 @@ public virtual IEdmNavigationSource ResolveNavigationSource(IEdmModel model, str
             }
 
             var result = container.Elements.OfType<IEdmNavigationSource>()
-                .Where(source => string.Equals(identifier, source.Name, StringComparison.OrdinalIgnoreCase)).ToList();
+                .Where(source => string.Equals(identifier, source.Name, StringComparison.OrdinalIgnoreCase));
+
+            IEdmNavigationSource resolvedNavigationSource = default(IEdmNavigationSource);

```

Comment:
>default(IEdmNavigationSource); [](start = 60, length = 30)

I don't understand what's the performance advantage to use "default(interface)" and "null"?

Reply:
To keep the parity. Previously, it was SingleOrDefault, to avoid any breaking changes, I relied on Default here rather than null. 

---
In reply to: [362107858](https://github.com/OData/odata.net/pull/1602#discussion_r362107858) [](ancestors = 362107858)